source/parser.hera

```diff
@@ -4164,7 +4164,9 @@ Statement
   KeywordStatement
   VariableStatement
   IfStatement !ShouldExpressionize -> $1
-  IterationStatement !ShouldExpressionize -> $1
+  IterationStatement !ShouldExpressionize ->
+    if ($1.star) return $skip
```

Comment:
Maybe add a comment for this line, like `// Leave starred forms for IterationExpression`